#!/usr/bin/env python

PACKAGE = 'diff_drive_controller'

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, bool_t, double_t

gen = ParameterGenerator()

# Kinematic parameters related
gen.add("left_wheel_radius_multiplier", double_t, 0, "Left wheel radius multiplier.", 1.0, 0.5, 1.5)
gen.add("right_wheel_radius_multiplier", double_t, 0, "Right wheel radius multiplier.", 1.0, 0.5, 1.5)
gen.add("wheel_separation_multiplier", double_t, 0, "Wheel separation multiplier.", 1.0, 0.5, 1.5)

# Acceleration, velocity, jerk parameters
gen.add("max_acceleration_z", double_t, 0, "Max acceleration.", 1.0, 0.1, 10.0)
gen.add("max_velocity", double_t, 0, "Max velocity.", 1.0, 0.1, 10.0)
gen.add("max_jerk", double_t, 0, "Max jerk.", 1.0, 0.1, 10.0)

# Publication related
gen.add("publish_rate", double_t, 0, "Publish rate of odom.", 50.0, 0.0, 2000.0)
gen.add("enable_odom_tf", bool_t, 0, "Publish odom frame to tf.", True)

exit(gen.generate(PACKAGE, "diff_drive_controller", "DiffDriveController"))
